test/common/stats/allocator_impl_test.cc

```diff
@@ -125,6 +127,142 @@ TEST_F(AllocatorImplTest, RefCountDecAllocRaceSynchronized) {
   EXPECT_FALSE(alloc_.isMutexLockedForTest());
 }
 
+TEST_F(AllocatorImplTest, ForEachSinkedCounter) {
+
+  StatNameHashSet sinked_stat_names;
+  std::vector<CounterSharedPtr> sinked_counters;
```

Comment:
it looks like these two vectors are written and never read in the test. Drop them?

Reply:
Don't we need at least one shared instance of the counters?